### PR TITLE
Files and changes for Allen Coral Atlas (ACA) v2 data

### PR DESCRIPTION
**Pull request by mitchest**
- added a new .jsonnet for the v2 dataset
- updates some v2 references
- added new .js example file

### CODE REVIEW

**Comment by mitchest** [comment]
reopening after signing CLA

File: catalog/ACA/ACA_reef_habitat_v2_0.jsonnet [hunk 1]
```diff
@@ -0,0 +1,320 @@
+local id = 'ACA/reef_habitat/v2_0';
```
Comment (simonff) [inline catalog/ACA/ACA_reef_habitat_v2_0.jsonnet:1]: could you add successor / predecessor / latest link to both old and new ACA entries similarly to the two files in https://github.com/google/earthengine-catalog/tree/main/catalog/CAS ?

File: examples/ACA_reef_habitat_v2_0.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+var dataset = ee.Image("ACA/reef_habitat/v2_0");
+
+// Teti'aroa, an atoll in French Polynesia
+Map.setCenter(-149.56194, -17.00872, 13);
+Map.setOptions('SATELLITE');
+
+// Mitch: The visualisations are baked into the image properties
```
Comment (simonff) [inline examples/ACA_reef_habitat_v2_0.js:7]: maybe remove "Mitch:" ?

File: catalog/ACA/ACA_reef_habitat_v2_0.jsonnet [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+        },
+      },
+    ],
+  },
+  'sci:doi': '10.5281/zenodo.3833242',
+  'sci:citation': |||
+    Allen Coral Atlas (2020). Imagery, maps and monitoring of the world's tropical coral reefs. Zenodo. [doi:10.5281/zenodo.3833242](https://doi.org/10.5281/zenodo.3833242)
+  |||,
+  'gee:terms_of_use': ee.gee_terms_of_use(license),
+  'gee:user_uploaded': true,
```
Comment (simonff) [inline catalog/ACA/ACA_reef_habitat_v2_0.jsonnet:319]: could you also add

'gee:skip_indexing': true

? It's a temporary thing we need to do for user-uploaded descriptions.

**Pull request closed without merging**